packages/pwa-kit-dev/src/configs/webpack/config.js

```diff
@@ -212,6 +203,60 @@ const withChunking = (config) => {
     }
 }
 
+const ruleForBabelLoader = (babelPlugins) => {
+    return {
+        test: /(\.js(x?)|\.ts(x?))$/,
+        exclude: /node_modules/,
+        use: [
+            {
+                loader: findInProjectThenSDK('babel-loader'),
+                options: {
+                    rootMode: 'upward',
+                    cacheDirectory: true,
+                    ...(babelPlugins ? {plugins: babelPlugins} : {})
+                }
+            }
+        ]
+    }
+}
+
+const enableReactRefresh = (config) => {
+    if (mode !== development) {
+        return config
+    }
+
+    const rules = [
+        ruleForBabelLoader([require.resolve('react-refresh/babel')]),
+        ...config.module.rules.slice(1)
```

Comment:
this seems fragile... can we give the loader rules an `id` and replace by `id`?

Reply:
@kevinxh I've tried adding an id and it failed because not matching the expected config schema. That's why I end up with this instead.

Reply:
This feels like the `Builder` class lacks some basic functionalities. We need to be able to manipulate the config in a safer manner. Imagine we have many extends functions and each of them adds/removes different rules, it's gonna be hard to keep the webpack config maintainable.